### PR TITLE
Add a wrapper header file precice.hpp

### PR DESCRIPTION
**Pull request by MakisH**
## Main changes of this PR

- Adds a very minimal `precice.hpp`, which only includes headers (`Participant.hpp`, `Version.h`, `span.hpp`).
- Renames the bindings files to `precicec.h` and `precicefortran.hpp`.

This probably has many side-effects. I have not checked the packaging or any downstream components.

To discuss:

- [x] Do we still want to include the additional headers (version, span) in the `Participant` files?

## Motivation and additional information

As discussed in https://github.com/precice/precice/issues/1458, we want to have a central header file `precice.hpp` to include the API and any auxiliary headers. The name is also more intuitive than `SolverInterface.hpp` or `Participant.hpp`.

Depends on https://github.com/precice/precice/pull/1643.

## Author's checklist

* [x] I used the [`pre-commit` hook](https://precice.org/dev-docs-dev-tooling.html#setting-up-pre-commit) to prevent dirty commits and used `pre-commit run --all` to format old commits.
* [x] I added a changelog file with `make changelog` if there are user-observable changes since the last release.
* I added a test to cover the proposed changes in our test suite. -> N/A
* [ ] For breaking changes: I documented the changes in the appropriate [porting guide](https://precice.org/couple-your-code-porting-overview.html).
* [x] I sticked to C++17 features.
* [x] I sticked to CMake version 3.16.3.
* [x] I squashed / am about to squash all commits that should be seen as one.

## Reviewers' checklist

* [x] Does the changelog entry make sense? Is it formatted correctly?
* [x] Do you understand the code changes?

### CODE REVIEW

File: src/precice/precice.hpp [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#pragma once
+
+#include <precice/Participant.hpp>
+#include <precice/Version.h>
+#include <precice/span.hpp>
```
Comment (fsimonis) [inline src/precice/precice.hpp:5]: Isn't this primarily an implementation detail of `Participant.hpp` similar to `export.hpp` and `types.hpp`?
If so, then this doesn't really belong here.

Comment (MakisH) [reply src/precice/precice.hpp:5]: But then, would the `precice.hpp` just be a redirection/alias for `Participant.hpp`?

Comment (MakisH) [reply src/precice/precice.hpp:5]: I thought we wanted to expose the `Version.h` and `span.hpp` to the user, to be able to use, e.g., the preCICE span directly in their code.

We can achieve this with either location, but I thought that putting them directly into the `precice.hpp` makes the intention clearer.

Comment (fsimonis) [reply src/precice/precice.hpp:5]: Key point here is that the user may have to create a `precice::span` in the adapter.
So, let's keep it in the `precice.hpp`.

Comment (MakisH) [reply src/precice/precice.hpp:5]: And same for `Version.h`, right? So, everything right, conversation resolved, and PR approved?

**Review by fsimonis — APPROVED** [review]

**Pull request merged**